src/material-experimental/mdc-slider/slider.ts

```diff
@@ -22,13 +75,229 @@ import {SliderAdapter} from './slider-adapter';
+
+  /** Whether the slider displays tick marks along the slider track. */
+  @Input()
+  get hasTickMarks(): boolean { return this._hasTickMarks; }
+  set hasTickMarks(v: boolean) { this._hasTickMarks = coerceBooleanProperty(this._hasTickMarks); }
+  private _hasTickMarks: boolean = false;
+
+  /** The minimum value that the slider can have. */
+  @Input()
+  get min(): number { return this._min; }
+  set min(v: number) { this._min = coerceNumberProperty(v, this._min); }
+  private _min = 0;
+
+  /** The maximum value that the slider can have. */
+  @Input()
+  get max(): number { return this._max; }
+  set max(v: number) { this._max = coerceNumberProperty(v, this._max); }
+  private _max = 100;
+
+  /** The values at which the thumb will snap. */
+  @Input()
+  get step(): number { return this._step; }
+  set step(v: number) { this._step = coerceNumberProperty(v, this._step); }
+  private _step: number = 1;
+
+  /**
+   * Function that will be used to format the value before it is displayed
+   * in the thumb label. Can be used to format very large number in order
+   * for them to fit into the slider thumb.
+   */
+  @Input() displayWith: ((value: number) => string) | null;
+
+  /** Event emitted when a slider thumb value has changed. */
+  @Output() readonly change: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /** Event emitted when a slider thumb moves. */
+  @Output() readonly input: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /** Event emitted when the slider thumb starts being dragged. */
+  @Output() readonly dragStart: EventEmitter<MatSliderInteraction>
+    = new EventEmitter<MatSliderInteraction>();
+
+  /** Event emitted when the slider thumb stops being dragged. */
+  @Output() readonly dragEnd: EventEmitter<MatSliderInteraction>
+    = new EventEmitter<MatSliderInteraction>();
+
+  /** Whether this is a ranged slider. */
+  get isRange(): boolean { return this._inputs.length === 2; }
 
-  // tslint:disable-next-line:no-unused-variable
+  /** Instance of the MDC slider foundation for this slider. */
   private _foundation = new MDCSliderFoundation(new SliderAdapter());
+
+  /** Whether the foundation has been initialized. */
+  _initialized: boolean = false;
+
+  /** The string representation of the start thumbs value. */
+  _startValueIndicatorText: string;
+
+  /** The string representation of the end thumbs value. */
+  _endValueIndicatorText: string;
+
+  /** Used to keep track of & render the active & inactive tick marks on the slider track. */
+  get tickMarks(): TickMark[] { return this._tickMarks; }
+  set tickMarks(v: TickMark[]) {
+    this._tickMarks = v;
+    this._cdr.markForCheck();
+  }
+  private _tickMarks: TickMark[];
+
+  constructor(
+    private _cdr: ChangeDetectorRef,
+    private readonly _elementRef: ElementRef<HTMLElement>,
+    private readonly _platform: Platform,
+    @Inject(DOCUMENT) private readonly _document: any) {}
```

Comment:
For `document`, follow the pattern used in other components so that the class member is correctly typed to `Document` while still using `any` for the constructor parameter:
```ts
private _document: Document;
constructor(@Inject(DOCUMENT) document: any) {
  this._document = document;
}
```